### PR TITLE
Vi/feat abstract class all relations

### PR DESCRIPTION
**Pull request by vincentvis**
Some improvement suggestions to the abstractService.

- allow relationships to be set to the findAll method (optional)
- allow creation and deletion through an array (create/delete many of T in one go)

Added tests for this through the Animal and changed the Animal to reflect those changes.

### CODE REVIEW

**Review by Ryno95 — CHANGES_REQUESTED** [review]
Definite improvement to have the Batch overload! Nice to have the relations as well ,  was thinking to do it tomorrow!

Maybe a good idea to add a e2e test for the new endpoints? I thinks it's easier testing as well  as the no need to manually set-up the validators like in the unit tests!

File: backend/src/test_example/animal.controller.ts [hunk 1]
```diff
@@ -31,13 +32,37 @@ export class AnimalController {
...
 
+	@Post('bulk')
+	async createBulk(
+		@Body(
+			new ParseArrayPipe({
+				items: CreateAnimalDto,
+				whitelist: true,
+				forbidNonWhitelisted: true,
+			}),
+		)
```
[the remark this answers is not in the document]
Comment (vincentvis) [reply backend/src/test_example/animal.controller.ts:47]: I tested quickly with curl, with and without extra stuff, and this is not done globally!
So `whitelist` and `forbidNonWhitelisted` are still needed here!

**Comment by vincentvis** [comment]
> Definite improvement to have the Batch overload! Nice to have the relations as well , was thinking to do it tomorrow!
> 
> Maybe a good idea to add a e2e test for the new endpoints? I thinks it's easier testing as well as the no need to manually set-up the validators like in the unit tests!

added e2e tests for bulk :)

File: backend/src/test_example/entities/animals.entity.ts [hunk 2]
```diff
@@ -6,5 +7,6 @@ export class AnimalEntity {
 	id: number;
 
 	@Column({ unique: true })
```
Comment (Ryno95) [inline backend/src/test_example/entities/animals.entity.ts:8]: `@IsNotEmpy `can be removed here, then  think we can merge!

Comment (Ryno95) [reply backend/src/test_example/entities/animals.entity.ts:8]: Already removed nice! ;D

**Review by Ryno95 — APPROVED** [review]

File: backend/src/test_example/entities/animals.entity.ts (hunk 2 above)
Comment (vincentvis) [reply backend/src/test_example/entities/animals.entity.ts:8]: done

**Pull request merged**